include/eve/algo/any_of.hpp

```diff
@@ -9,22 +9,30 @@
 
 #include <eve/algo/array_utils.hpp>
 #include <eve/algo/for_each_iteration.hpp>
-#include <eve/algo/one_range_algorithm_adapter.hpp>
+#include <eve/algo/preprocess_range.hpp>
 #include <eve/algo/traits.hpp>
 
 #include <eve/function/any.hpp>
 #include <eve/function/logical_or.hpp>
 
 #include <array>
+#include <utility>
 
 namespace eve::algo
 {
-  struct any_of_ : one_range_algorithm_adapter<any_of_>
+  template <instance_of<algo::traits> Traits>
```

Comment:
where is instance_of defined ? Is it an old concept we have already ?

Reply:
We do. It's in https://github.com/jfalcou/eve/blob/73c6ea8612bf0e975446b351ac94d8631cd3228c/include/eve/algo/concepts.hpp#L110

Really need it a lot.

Reply:
Great